doc/user/content/guides/temporal-filters.md

```diff
@@ -22,6 +22,14 @@ You can only use `mz_logical_timestamp()` to establish a temporal filter in the
 
 At the moment, you can't use the `!=` operator with `mz_logical_timestamp` (we're working on it).
 
+## Late-arriving data
+
+Records that arrive late (that is, with a timestamp that is too old to pass the filter) are not shown in the current window. Think of a temporal filter as being like a query to a database for records updated between certain timestamps: if the record is added to the database with a time before the earlier timestamp boundary, it will not appear in the query results.
```

Comment:
What does "current window" mean here? I would say records that arrive late are not show (they do not satisfy the predicate on their arrival, and will not in the future satisfy it).

I think the conceptual model for a "temporal filter" is perhaps much simpler than all of this. The query is re-evaluated moment by moment, and we keep the answers up to date. If a record is added at a `time` such that for all future `mz_logical_timestamp()` times (those greater or equal to `time`) the predicate is not satisfied, then the record does not appear. This is not because of weirdness of temporal filters, but just a consequence of "this query executed moment by moment will never present this record".